pkg/mgmt/lvmnode/lvmnode.go

```diff
@@ -142,17 +163,21 @@ func (c *NodeController) updateNode(oldObj, newObj interface{}) {
 
 // deleteNode is the delete event handler for LVMNode
 func (c *NodeController) deleteNode(obj interface{}) {
-	node, ok := obj.(*apis.LVMNode)
+	node, ok := c.getStructuredObject(obj)
 	if !ok {
-		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-		if !ok {
-			runtime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
-			return
-		}
-		node, ok = tombstone.Obj.(*apis.LVMNode)
-		if !ok {
-			runtime.HandleError(fmt.Errorf("Tombstone contained object that is not a LVMNode %#v", obj))
-			return
+		unstructuredObj, ok := obj.(*unstructured.Unstructured)
+		if ok {
```

Comment:
What the `!ok` case? Shouldn't we return error in case of this?

Reply:
Incorporated